tests/fixture/tmpnet/node_process.go

```diff
@@ -0,0 +1,258 @@
+	}
+
+	go func() {
+		if err := cmd.Wait(); err != nil {
+			if err.Error() != "signal: killed" {
+				_, _ = fmt.Fprintf(w, "%s finished with error: %v\n", nodeDescription, err)
+			}
+		}
+		_, _ = fmt.Fprintf(w, "%s exited\n", nodeDescription)
+	}()
+
+	// A node writes a process context file on start. If the file is not
+	// found in a reasonable amount of time, the node is unlikely to have
+	// started successfully.
+	if err := p.waitForProcessContext(context.Background()); err != nil {
+		return fmt.Errorf("failed to start local node: %w", err)
+	}
+
+	_, err = fmt.Fprintf(w, "Started %s\n", nodeDescription)
+	return err
+}
+
+// Signals the node process to stop.
+func (p *NodeProcess) InitiateStop() error {
+	proc, err := p.getProcess()
+	if err != nil {
+		return fmt.Errorf("failed to retrieve process to stop: %w", err)
+	}
+	if proc == nil {
+		// Already stopped
+		return nil
+	}
+	if err := proc.Signal(syscall.SIGTERM); err != nil {
+		return fmt.Errorf("failed to send SIGTERM to pid %d: %w", p.pid, err)
+	}
+	return nil
+}
+
+// Waits for the node process to stop.
+func (p *NodeProcess) WaitForStopped(ctx context.Context) error {
+	ticker := time.NewTicker(defaultNodeTickerInterval)
+	defer ticker.Stop()
+	for {
+		proc, err := p.getProcess()
+		if err != nil {
+			return fmt.Errorf("failed to retrieve process: %w", err)
+		}
+		if proc == nil {
+			return nil
+		}
+
+		select {
+		case <-ctx.Done():
+			return fmt.Errorf("failed to see node process stop %q before timeout: %w", p.node.ID, ctx.Err())
+		case <-ticker.C:
+		}
+	}
+}
+
+func (p *NodeProcess) IsHealthy(ctx context.Context) (bool, error) {
+	// Check that the node process is running as a precondition for
+	// checking health. getProcess will also ensure that the node's
+	// API URI is current.
+	proc, err := p.getProcess()
+	if err != nil {
+		return false, fmt.Errorf("failed to determine process status: %w", err)
+	}
+	if proc == nil {
+		return false, ErrNotRunning
+	}
+
+	return checkNodeHealth(ctx, p.node.URI)
+}
+
+func (p *NodeProcess) getProcessContextPath() string {
+	return filepath.Join(p.node.getDataDir(), config.DefaultProcessContextFilename)
+}
+
+func (p *NodeProcess) waitForProcessContext(ctx context.Context) error {
+	ticker := time.NewTicker(defaultNodeTickerInterval)
+	defer ticker.Stop()
+
+	ctx, cancel := context.WithTimeout(ctx, defaultNodeInitTimeout)
+	defer cancel()
+	for len(p.node.URI) == 0 {
+		err := p.readState()
+		if err != nil {
+			return fmt.Errorf("failed to read process context for node %q: %w", p.node.ID, err)
+		}
+
+		select {
+		case <-ctx.Done():
+			return fmt.Errorf("failed to load process context for node %q before timeout: %w", p.node.ID, ctx.Err())
+		case <-ticker.C:
+		}
+	}
+	return nil
+}
+
+// Retrieve the node process if it is running. As part of determining
+// process liveness, the node's process context will be refreshed if
+// live or cleared if not running.
+func (p *NodeProcess) getProcess() (*os.Process, error) {
```

Comment:
When we do `exec.Command` to start the binary, we have a reference to the `Process`. I think we can use that instead, right?